### PR TITLE
Detemplated one of the module solutions

### PR DESCRIPTION
**Pull request by SansPapyrus683**

### CODE REVIEW

File: content/6_Advanced/Comb_Sub.mdx [hunk 1]
```diff
@@ -86,39 +86,62 @@ Similar to the editorial, define $\texttt{dp[x][0][g]}$ to be the minimum cost t
...
 ```cpp
-int n,b,c[5001],d[5001];
-vl dp[5001][2];
-vi child[5001];
-
-vl comb(vl a, vl b) {
-	vl c(sz(a)+sz(b)-1,INF);
-	F0R(i,sz(a)) F0R(j,sz(b)) ckmin(c[i+j],a[i]+b[j]);
-	return c;
+constexpr int MAX_GOODS = 5000;
```
Comment (bqi343) [inline content/6_Advanced/Comb_Sub.mdx:89]: make code compile on its own

File: content/6_Advanced/Comb_Sub.mdx [hunk 2]
```diff
@@ -86,39 +86,62 @@ Similar to the editorial, define $\texttt{dp[x][0][g]}$ to be the minimum cost t
...
+constexpr long long INF = 1e18;
+
+int initial[MAX_GOODS + 1];
+int discounted[MAX_GOODS + 1];
+vector<long long> dp[MAX_GOODS + 1][2];
+vector<int> child[MAX_GOODS + 1];
+
+vector<long long> total_min(vector<long long> a, vector<long long> b) {
+	vector<long long> combined(a.size() + b.size() -1, INF);
```
Comment (bqi343) [inline content/6_Advanced/Comb_Sub.mdx:98]: `- 1`?

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/Comb_Sub.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply content/6_Advanced/Comb_Sub.mdx:89]: So just add all of the `#include`s and that stuff?

Comment (bqi343) [reply content/6_Advanced/Comb_Sub.mdx:89]: yea

**Review by dongliu0426 — APPROVED** [review]

File: content/6_Advanced/Comb_Sub.mdx [hunk 3]
```diff
@@ -86,39 +86,67 @@ Similar to the editorial, define $\texttt{dp[x][0][g]}$ to be the minimum cost t
...
+	/*
+	 * we have INF for the first element because 0 would mess up merging
+	 * if it WAS 0, then that would mean we didn't buy this one
+	 * yet the algo would still think all the children are eligible for coupons
+	 */
+	dp[g][1] = {INF, discounted[g]};
+	for (int t : child[g]) {
+		process(t);
+		dp[g][0] = total_min(dp[g][0],dp[t][0]);
+		dp[g][1] = total_min(dp[g][1],dp[t][1]);
```
Comment (dongliu0426) [inline content/6_Advanced/Comb_Sub.mdx:123]:
```suggestion
		dp[g][0] = total_min(dp[g][0], dp[t][0]);
		dp[g][1] = total_min(dp[g][1], dp[t][1]);
```

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

**Comment by dongliu0426** [comment]
oops my bad

**Pull request merged**